apps/channels/tests/test_facebook_integration.py

```diff
@@ -1,93 +1,87 @@
-        message = facebook_messages.text_message(self.page_id, message="Hi there")
-        handle_facebook_message(team_slug=self.team.slug, message_data=message)
-        called_args, called_kwargs = new_user_message.call_args
-        facebook_message = called_args[0]
-        assert facebook_message.page_id == self.page_id
-        assert facebook_message.message_text == "Hi there"
-        assert facebook_message.content_type == MESSAGE_TYPES.TEXT
-        assert facebook_message.user_id == "6785984231"
-        assert facebook_message.media_url is None
 
-    @patch("apps.channels.tasks.FacebookMessengerChannel.new_user_message")
-    def test_incoming_voice_message(self, new_user_message):
-        """Verify that a FacebookMessage object is being built correctly for voice messages"""
-        media_url = "https://example.com/my-audio"
-        message = facebook_messages.audio_message(self.page_id, attachment_url=media_url)
-        handle_facebook_message(team_slug=self.team.slug, message_data=message)
-        called_args, called_kwargs = new_user_message.call_args
-        facebook_message = called_args[0]
-        assert facebook_message.page_id == self.page_id
-        assert facebook_message.message_text == ""
-        assert facebook_message.content_type == MESSAGE_TYPES.VOICE
-        assert facebook_message.user_id == "6785984231"
-        assert facebook_message.media_url == media_url
+class TestTwilio:
+    @pytest.mark.parametrize(
+        ("message", "message_type"),
+        [
+            (twilio_messages.Messenger.text_message(), "text"),
+            (twilio_messages.Messenger.audio_message(), "voice"),
+        ],
+    )
+    def test_parse_messages(self, message, message_type):
+        whatsapp_message = TwilioMessage.parse(json.loads(message))
+        assert whatsapp_message.platform == ChannelPlatform.FACEBOOK
+        assert whatsapp_message.chat_id == whatsapp_message.from_
+        if message_type == "text":
+            assert whatsapp_message.content_type == MESSAGE_TYPES.TEXT
+            assert whatsapp_message.media_url is None
+        else:
+            assert whatsapp_message.content_type == MESSAGE_TYPES.VOICE
+            assert whatsapp_message.media_url == "http://example.com/media"
+
+    @pytest.mark.usefixtures("_twilio_whatsapp_channel")
+    @pytest.mark.parametrize(
+        ("incoming_message", "message_type"),
+        [
+            (twilio_messages.Messenger.text_message(), "text"),
+            (twilio_messages.Messenger.audio_message(), "audio"),
+        ],
+    )
+    @patch("apps.service_providers.speech_service.SpeechService.synthesize_voice")
+    @patch("apps.chat.channels.ChannelBase._get_voice_transcript")
+    @patch("apps.service_providers.messaging_service.TwilioService.send_text_message")
+    @patch("apps.chat.channels.FacebookMessengerChannel._get_llm_response")
+    def test_twilio_uses_facebook_channel_implementation(
+        self,
+        get_llm_response_mock,
```

Comment:
Should this also assert that `get_llm_response_mock` get's called?

Reply:
We don't really have to, since it must have been called when `send_text_message` was called. I'll in anycase just update the code to assert stuff when voice was used. I see this is missing